core/trino-main/src/main/java/io/trino/execution/TaskManagerConfig.java

```diff
@@ -385,6 +387,34 @@ public TaskManagerConfig setInfoMaxAge(Duration infoMaxAge)
         return this;
     }
 
+    public boolean isScaleWritersEnabled()
+    {
+        return scaleWritersEnabled;
+    }
+
+    @Config("task.scale-writers-enabled")
+    @ConfigDescription("Scale the number of concurrent table writers per task based on throughput")
+    public TaskManagerConfig setScaleWritersEnabled(boolean scaleWritersEnabled)
+    {
+        this.scaleWritersEnabled = scaleWritersEnabled;
+        return this;
+    }
+
+    @Min(1)
+    @PowerOfTwo
```

Comment:
Do we need to make it power of two ? At least in the current code I didn't see a problem with allowing different numbers.

Reply:
Yeah, it's not needed for now since we are using fast mod only in the case of partitioned data. Although, let's keep it this way as we have plans to extend for partition use-case.